### PR TITLE
_user/voxeldemo: Add Voxel Space example gfx demo

### PR DESCRIPTION
**Pull request by gerard5**
https://user-images.githubusercontent.com/141153/122703660-62b34880-d252-11eb-8efb-51855b236c94.mp4

JIRA: RTOS-49

## Description

Voxel Space demo which is self contained and does not depend on external resources voxel terrain is procedurally generated.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (ia32-generic).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: _user/voxeldemo/main.c [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+	return n;
+}
+
+
+void voxel_free(voxel_t *v)
+{
+	free(v->bufBack);
+	free(v->mapColor);
+	free(v->mapHeight);
+}
```
Comment (kemonats) [inline _user/voxeldemo/main.c:69]:
```suggestion
	free(v->mapHeight);
	free(v->palette);
}
```

File: _user/voxeldemo/main.c [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+
+int voxel_demo(graph_t *g)
+{
+	unsigned int t = 0;
+	voxel_t v = {
+		.width = g->width,
+		.height = g->height,
+		.pixels = g->data,
+		.cam = { .h = 300, .horiz = 200, .dist = 300 },
+	};
```
Comment (kemonats) [inline _user/voxeldemo/main.c:305]:
```suggestion
		.cam = { .h = 300, .horiz = 200, .dist = 300 },
		.bufBack = NULL;
		.mapHeight = NULL;
		.mapColor = NULL;
		.palette = NULL;			
	};
```

**Comment by pawelpisarczyk** [comment]
Demo should work on any target. Default adapter should be used. Now it's hardcoded to use virtio-gpu.

File: _user/voxeldemo/main.c (hunk 1 above)
Comment (gerard5) [reply _user/voxeldemo/main.c:69]: Yes, `palette` is not freed, this is to be fixed.

File: _user/voxeldemo/main.c (hunk 2 above)
Comment (gerard5) [reply _user/voxeldemo/main.c:305]: It is not necessary. Any structure element in a C initializer that is not listed defaults to `0` / `NULL`.

**Pull request merged**